pkg/gadgets/seccomp/gadget.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package seccomp
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"runtime"
+	"sort"
+	"sync"
+	"syscall"
+
+	"github.com/cilium/ebpf"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	libseccomp "github.com/seccomp/libseccomp-golang"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/types"
+
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/api/v1alpha1"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets"
+)
+
+// #include "../../../gadget-container/gadgets/seccomp/bpf/common.h"
+import "C"
+
+const (
+	BPF_PROG_NAME = "tracepoint__raw_syscalls__sys_enter"
+	BPF_MAP_NAME  = "syscalls_per_mntns"
+)
+
+type Trace struct {
```

Comment:
The Trace/TraceFactory code is almost identical in the different gadgets that use it... Is there something preventing us from having some common code that implements this rather than repeating it in each gadget?

Reply:
I agree we should have common code. It should be doable and it would benefit other gadgets.

For the seccomp gadget specifically, this should be reworked differently because the tracer should be started only one time and it can be queried from different Trace custom resource. So some state should be moved from the `Trace struct` to the `TraceFactory` struct somehow.

Other gadgets like biolatency, opensnoop, execsnoop have one tracer running for each `Tracer` custom resource, so that's a different model.

So I am not sure how this common code would look like yet.